### PR TITLE
Refactor MTE-1938 [v123] Move firefox application files into firefox-ios folder 

### PR DESCRIPTION
**Pull request by isabelrios**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-1938)

There are only 6036 files changed 😓  althought most of them are just renaming as they are moved into firefox-ios folder.

I would like to have an idea about the main changes needed so that CI works. That's why I'm opening this PR as draft to iterate over it.

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by isabelrios** [comment]
@OrlaM @lmarceau Looks like I could successfully take out BrowserKit from firefox-ios folder. It is in the root directory now. Bitrise is green to build and run unit + ui tests. Could you please give a first pass to this huge PR? 
Not sure the best way, but similar to the browser kit change, there may be others I have not taken into account...

Thank you!! 🙏

File: BrowserKit/Sources/WebEngine/Engine.swift [hunk 1]
```diff
@@ -1,10 +0,0 @@
-// This Source Code Form is subject to the terms of the Mozilla Public
-// License, v. 2.0. If a copy of the MPL was not distributed with this
-// file, You can obtain one at http://mozilla.org/MPL/2.0/
-
-import Foundation
```
Comment (lmarceau) [inline BrowserKit/Sources/WebEngine/Engine.swift:5]: I added those files in `BrowserKit/Sources/WebEngine` in a recent PR, I guess with the move it's getting removed but they should be put back.

File: firefox-ios/Client/Frontend/Browser/Tabs/Models/TabModel.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+struct TabModel: Equatable {
```
Comment (lmarceau) [inline firefox-ios/Client/Frontend/Browser/Tabs/Models/TabModel.swift:7]: I don't know why this file appears as if it's added? It's in main as well, but shouldn't it be just changing location as the other files? There's a couple of other files like this (TabsState.swift, TabsDisplayPanel. TabTrayViewController...)

**Review by lmarceau — COMMENTED** [review]
2 comments from a coding perspective!

File: BrowserKit/Sources/WebEngine/Engine.swift (hunk 1 above)
Comment (isabelrios) [reply BrowserKit/Sources/WebEngine/Engine.swift:5]: I will add those back...

File: firefox-ios/Client/Frontend/Browser/Tabs/Models/TabModel.swift (hunk 2 above)
Comment (isabelrios) [reply firefox-ios/Client/Frontend/Browser/Tabs/Models/TabModel.swift:7]: you are right, they should not appear as new files. I have seen the same with other files and was able to fix it. I will do the same with these

File: firefox-ios/Client/Frontend/Browser/Tabs/State/TabsState.swift [hunk 3]
```diff
@@ -0,0 +1,69 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (isabelrios) [inline firefox-ios/Client/Frontend/Browser/Tabs/State/TabsState.swift:1]: This file is not in main, I'm removing it

File: firefox-ios/Client/Frontend/Browser/Tabs/Models/TabModel.swift (hunk 2 above)
Comment (isabelrios) [reply firefox-ios/Client/Frontend/Browser/Tabs/Models/TabModel.swift]: Fixed, hopefully it is not added after fixing conflicts again 😅

File: firefox-ios/Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 4]
```diff
@@ -128,4 +128,4 @@
     }
   ],
   "version" : 2
-}
+}
```
Comment (brennie) [inline firefox-ios/Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:131]: Missing newline at EOF

**Comment by brennie** [comment]
I see that nimbus.fml.yaml is moved into the firefox-ios directory, but there is also a new top-level nimbus.fml.yaml (in your fork here: https://github.com/isabelrios/firefox-ios/blob/irios-mte1938-move-application-files-into-folder/nimbus.fml.yaml). This file shouldn't exist. (I could not comment on the file directly because the file view is very unresponsive with a PR of this size).

**Review by flodolo — COMMENTED** [review]
From a quick look, I believe you need to update the paths in the workflow to import translations: https://github.com/mozilla-mobile/firefox-ios/blob/main/.github/workflows/firefox-ios-import-strings.yml#L52-L53

Looks like bootstrap and import-strings.sh remain in the same place, but this line should also change: https://github.com/isabelrios/firefox-ios/blob/irios-mte1938-move-application-files-into-folder/firefox-ios/import-strings.sh#L8

**Comment by isabelrios** [comment]
> I see that nimbus.fml.yaml is moved into the firefox-ios directory, but there is also a new top-level nimbus.fml.yaml (in your fork here: https://github.com/isabelrios/firefox-ios/blob/irios-mte1938-move-application-files-into-folder/nimbus.fml.yaml). This file shouldn't exist. (I could not comment on the file directly because the file view is very unresponsive with a PR of this size).

Thanks, I will remove the one in the root directory then

File: WidgetKit/uz.lproj/WidgetIntents.strings
Comment (isabelrios) [inline WidgetKit/uz.lproj/WidgetIntents.strings:1]: Actually I can see this file in the new location firefox-ios/WidgetKit/uz.lproj/WidgetIntents.strings.. I will investigate and fix this

**Comment by isabelrios** [comment]
Bitrise is green with the latest commit [98d230b](https://github.com/mozilla-mobile/firefox-ios/pull/17460/commits/98d230b828d3bc28eab27617897efa6d309bf98e): https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/f19e933e-3c7f-4795-8462-4bcd2c932988

**Comment by isabelrios** [comment]
@brennie once this PR lands, there might be changes needed in the nimbus scripts... I had to add in this folder those files to get the project compiling https://github.com/isabelrios/firefox-ios/tree/irios-mte1938-move-application-files-into-folder/firefox-ios/bin 
Hope that is fine and we have follow up PRs to modify what's needed?

**Comment by brennie** [comment]
> @brennie once this PR lands, there might be changes needed in the nimbus scripts... I had to add in this folder those files to get the project compiling https://github.com/isabelrios/firefox-ios/tree/irios-mte1938-move-application-files-into-folder/firefox-ios/bin Hope that is fine and we have follow up PRs to modify what's needed?

cc @jhugman -- is it ok that these scripts moved?

Otherwise, I think everything should be a-ok on our end.

File: bootstrap.sh [hunk 5]
```diff
@@ -38,7 +38,7 @@ if [ "$1" == "--importLocales" ]; then
 fi
 
 # Download the nimbus-fml.sh script from application-services.
-NIMBUS_FML_FILE=./nimbus.fml.yaml
+NIMBUS_FML_FILE=./firefox-ios/nimbus.fml.yaml
```
Comment (jhugman) [inline bootstrap.sh:41]: The nimbus bootstrap script downloads the nimbus-fml.sh to the `$(pwd)/bin`. You can either wrap the `curl | bash`  with a `pushd firefox-ios`/`popd` combination (that changes the directory), or you can pass a `--directory` option to the script:

```sh
curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/mozilla/application-services/main/components/nimbus/ios/scripts/bootstrap.sh \
    | bash -s -- --directory ./firefox-ios/bin $NIMBUS_FML_FILE
```

Comment (jhugman) [reply bootstrap.sh:41]: The firefox-ios `project.projpbx` calls the `$(SOURCE_DIR)/bin/nimbus-fml.sh`, so we need to have `nimbus-fml.sh` in the `bin` directory relative to the `firefox-ios` directory.

The nimbus bootstrap downloads the `nimbus-fml.sh`, so as long as we're able to run the fxios `bootstrap.sh`, you can remove the repository version of `nimbus-fml.sh`.

The nimbus bootstrap also tries to make a `.gitignore` in the current directory, which you will probably want to check in.

In summary, I think from a nimbus point of view: 

1. You should change the fxios `bootstrap.sh` script as above, 
2. move the contents of `./bin/nimbus*.sh` into `./firefox-ios/bin` 
3. remove `nimbus-fml.sh` (and check the deletion in to the git)
4. check in the newly formed `.gitignore` file in to git.

As for the rest of it: this is a mammoth under-taking. Good job diving in and doing it! Thank you so much!

**Review by OrlaM — APPROVED** [review]

File: .github/workflows/firefox-ios-import-strings.yml [hunk 6]
```diff
@@ -49,8 +49,8 @@ jobs:
       run: sh ./bootstrap.sh --importLocales
     - name: Update new strings
       run: |-
-        git diff || (git add Shared/*/*.lproj/* Shared/*.lproj/* WidgetKit/*.lproj/* Client/*/*.lproj/* Client/*.lproj/*)
-        git restore Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
+        git diff || (git add firefox-ios/Shared/*/*.lproj/* firefox-ios/Shared/*.lproj/* firefox-ios/WidgetKit/*.lproj/* firefox-ios/Client/*/*.lproj/* firefox-ios/Client/*.lproj/*)
+        git restore firefox-ios/Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
     - name: Create Pull Request
       uses: peter-evans/create-pull-request@v4
```
Comment (flodolo) [inline .github/workflows/firefox-ios-import-strings.yml:55]:
```suggestion
      uses: peter-evans/create-pull-request@v5
```

There is at least a `Buffer()` deprecation fixed in v5.

Also checkout is available in v4 (currently v3), setup-python in v5 (currently v3)

**Review by dragosb01 — APPROVED** [review]

**Review by jjSDET — APPROVED** [review]
LGTM
